### PR TITLE
Emoji shortcode

### PR DESCRIPTION
**Pull request by ralsina**
### Pull Request Checklist

- [x] I’ve read the [guidelines for contributing](https://github.com/getnikola/nikola/blob/master/CONTRIBUTING.rst).
- [x] I updated AUTHORS.txt and CHANGES.txt (if the change is non-trivial) and documentation (if applicable).
- [x] I tested my changes.

### Description

A simple shortcode for emoji support: {{% emoji cry_face %}}

### CODE REVIEW

File: nikola/plugins/shortcode/emoji/__init__.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
+# -*- coding: utf-8 -*-
+# This file is public domain according to its author, Brian Hsu
```
Comment (Kwpolska) [inline nikola/plugins/shortcode/emoji/__init__.py]: leftover 1

File: nikola/plugins/shortcode/emoji/__init__.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+from nikola.plugin_categories import ShortcodePlugin
+from nikola import utils
+
+TABLE = {}
+
+LOGGER = utils.get_logger('scan_posts', utils.STDERR_HANDLER)
+
+
+class Plugin(ShortcodePlugin):
+    """Plugin for gist directive."""
```
Comment (Kwpolska) [inline nikola/plugins/shortcode/emoji/__init__.py]: leftover 2

**Comment by Kwpolska** [comment]
Looks good, but perhaps we could merge all those JSON files into a more compact, manageable format of `{name: icon}`?

**Comment by ralsina** [comment]
@Kwpolska yeah, I could merge them but then I would need to do it again if I ever refresh that data.

**Comment by Kwpolska** [comment]
@ralsina I didn’t expect you to do it by hand.

File: nikola/plugins/shortcode/emoji/data/LICENSE [hunk 3]
```diff
@@ -0,0 +1,21 @@
+The MIT License (MIT)
+
+Copyright (c) 2016 -2017 Shayan Rais
```
Comment (Kwpolska) [inline nikola/plugins/shortcode/emoji/data/LICENSE]: Add a link to the [repo](https://github.com/shanraisshan/EmojiCodeSheet) somewhere.

File: nikola/plugins/shortcode/emoji/__init__.py [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+    def handler(self, name, filename=None, site=None, data=None, lang=None, post=None):
+        """Create HTML for emoji."""
+        if not TABLE:
+            self._populate()
+        try:
+            output = '''<span class="emoji">{}</span>'''.format(TABLE[name])
+        except KeyError:
+            LOGGER.warning('Unknown emoji {}'.format(name))
+            output = '''<span class="emoji, error">{}</span>'''.format(name)
```
Comment (Kwpolska) [inline nikola/plugins/shortcode/emoji/__init__.py]: No comma here.

**Pull request merged**